### PR TITLE
Added cpu interrupts, softirqs, softnet-stat

### PR DESCRIPTION
**Pull request by yuri-zubov**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

- [ ] Tests

- [x] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### TEST Snippet
```
bundle exec metrics-cpu-interrupts.rb
yuri.cpu.IO-APIC.2-edge.timer.0.CPU0 9 1523264148
yuri.cpu.IO-APIC.2-edge.timer.0.CPU1 0 1523264148
yuri.cpu.IO-APIC.2-edge.timer.0.CPU2 0 1523264148
yuri.cpu.IO-APIC.2-edge.timer.0.CPU3 0 1523264148
yuri.cpu.IO-APIC.1-edge.i8042.1.CPU0 19 1523264148
yuri.cpu.IO-APIC.1-edge.i8042.1.CPU1 0 1523264148
yuri.cpu.IO-APIC.1-edge.i8042.1.CPU2 0 1523264148
yuri.cpu.IO-APIC.1-edge.i8042.1.CPU3 0 1523264148
yuri.cpu.IO-APIC.8-edge.rtc0.8.CPU0 0 1523264148
yuri.cpu.IO-APIC.8-edge.rtc0.8.CPU1 1 1523264148
yuri.cpu.IO-APIC.8-edge.rtc0.8.CPU2 0 1523264148
yuri.cpu.IO-APIC.8-edge.rtc0.8.CPU3 0 1523264148
yuri.cpu.IO-APIC.9-fasteoi.acpi.9.CPU0 0 1523264148
yuri.cpu.IO-APIC.9-fasteoi.acpi.9.CPU1 1123465 1523264148
yuri.cpu.IO-APIC.9-fasteoi.acpi.9.CPU2 0 1523264148
yuri.cpu.IO-APIC.9-fasteoi.acpi.9.CPU3 0 1523264148
yuri.cpu.IO-APIC.12-edge.i8042.12.CPU0 0 1523264148
yuri.cpu.IO-APIC.12-edge.i8042.12.CPU1 0 1523264148
yuri.cpu.IO-APIC.12-edge.i8042.12.CPU2 0 1523264148
yuri.cpu.IO-APIC.12-edge.i8042.12.CPU3 220 1523264148
yuri.cpu.IO-APIC.94-edge.lis3lv02d.94.CPU0 0 1523264148
yuri.cpu.IO-APIC.94-edge.lis3lv02d.94.CPU1 0 1523264148
yuri.cpu.IO-APIC.94-edge.lis3lv02d.94.CPU2 0 1523264148
yuri.cpu.IO-APIC.94-edge.lis3lv02d.94.CPU3 0 1523264148
yuri.cpu.PCI-MSI.458752-edge.aerdrv,.PCIe.PME.120.CPU0 0 1523264148
yuri.cpu.PCI-MSI.458752-edge.aerdrv,.PCIe.PME.120.CPU1 0 1523264148
yuri.cpu.PCI-MSI.458752-edge.aerdrv,.PCIe.PME.120.CPU2 0 1523264148
yuri.cpu.PCI-MSI.458752-edge.aerdrv,.PCIe.PME.120.CPU3 0 1523264148
yuri.cpu.PCI-MSI.466944-edge.aerdrv,.PCIe.PME.121.CPU0 0 1523264148
yuri.cpu.PCI-MSI.466944-edge.aerdrv,.PCIe.PME.121.CPU1 0 1523264148
yuri.cpu.PCI-MSI.466944-edge.aerdrv,.PCIe.PME.121.CPU2 3 1523264148
yuri.cpu.PCI-MSI.466944-edge.aerdrv,.PCIe.PME.121.CPU3 1 1523264148
yuri.cpu.PCI-MSI.468992-edge.aerdrv,.PCIe.PME.122.CPU0 0 1523264148
yuri.cpu.PCI-MSI.468992-edge.aerdrv,.PCIe.PME.122.CPU1 0 1523264148
yuri.cpu.PCI-MSI.468992-edge.aerdrv,.PCIe.PME.122.CPU2 0 1523264148
yuri.cpu.PCI-MSI.468992-edge.aerdrv,.PCIe.PME.122.CPU3 0 1523264148
yuri.cpu.PCI-MSI.475136-edge.aerdrv,.PCIe.PME,.pciehp.123.CPU0 0 1523264148
yuri.cpu.PCI-MSI.475136-edge.aerdrv,.PCIe.PME,.pciehp.123.CPU1 0 1523264148
yuri.cpu.PCI-MSI.475136-edge.aerdrv,.PCIe.PME,.pciehp.123.CPU2 0 1523264148
yuri.cpu.PCI-MSI.475136-edge.aerdrv,.PCIe.PME,.pciehp.123.CPU3 0 1523264148
yuri.cpu.PCI-MSI.327680-edge.xhci_hcd.124.CPU0 0 1523264148
yuri.cpu.PCI-MSI.327680-edge.xhci_hcd.124.CPU1 0 1523264148
yuri.cpu.PCI-MSI.327680-edge.xhci_hcd.124.CPU2 125800 1523264148
yuri.cpu.PCI-MSI.327680-edge.xhci_hcd.124.CPU3 2 1523264148
yuri.cpu.PCI-MSI.376832-edge.ahci[0000:00:17.0].125.CPU0 0 1523264148
yuri.cpu.PCI-MSI.376832-edge.ahci[0000:00:17.0].125.CPU1 0 1523264148
yuri.cpu.PCI-MSI.376832-edge.ahci[0000:00:17.0].125.CPU2 35015 1523264148
yuri.cpu.PCI-MSI.376832-edge.ahci[0000:00:17.0].125.CPU3 454242 1523264148
yuri.cpu.PCI-MSI.2097152-edge.rtsx_pci.126.CPU0 0 1523264148
yuri.cpu.PCI-MSI.2097152-edge.rtsx_pci.126.CPU1 0 1523264148
yuri.cpu.PCI-MSI.2097152-edge.rtsx_pci.126.CPU2 0 1523264148
yuri.cpu.PCI-MSI.2097152-edge.rtsx_pci.126.CPU3 22 1523264148
yuri.cpu.PCI-MSI.1048576-edge.enp2s0.127.CPU0 268 1523264148
yuri.cpu.PCI-MSI.1048576-edge.enp2s0.127.CPU1 8177988 1523264148
yuri.cpu.PCI-MSI.1048576-edge.enp2s0.127.CPU2 1480 1523264148
yuri.cpu.PCI-MSI.1048576-edge.enp2s0.127.CPU3 55 1523264148
yuri.cpu.PCI-MSI.32768-edge.i915.128.CPU0 1455 1523264148
yuri.cpu.PCI-MSI.32768-edge.i915.128.CPU1 607 1523264148
yuri.cpu.PCI-MSI.32768-edge.i915.128.CPU2 0 1523264148
yuri.cpu.PCI-MSI.32768-edge.i915.128.CPU3 0 1523264148
yuri.cpu.PCI-MSI.360448-edge.mei_me.129.CPU0 0 1523264148
yuri.cpu.PCI-MSI.360448-edge.mei_me.129.CPU1 0 1523264148
yuri.cpu.PCI-MSI.360448-edge.mei_me.129.CPU2 37 1523264148
yuri.cpu.PCI-MSI.360448-edge.mei_me.129.CPU3 0 1523264148
yuri.cpu.PCI-MSI.1572864-edge.iwlwifi.130.CPU0 14265 1523264148
yuri.cpu.PCI-MSI.1572864-edge.iwlwifi.130.CPU1 0 1523264148
yuri.cpu.PCI-MSI.1572864-edge.iwlwifi.130.CPU2 5695 1523264148
yuri.cpu.PCI-MSI.1572864-edge.iwlwifi.130.CPU3 1519 1523264148
yuri.cpu.PCI-MSI.514048-edge.snd_hda_intel:card0.131.CPU0 1144 1523264148
yuri.cpu.PCI-MSI.514048-edge.snd_hda_intel:card0.131.CPU1 0 1523264148
yuri.cpu.PCI-MSI.514048-edge.snd_hda_intel:card0.131.CPU2 0 1523264148
yuri.cpu.PCI-MSI.514048-edge.snd_hda_intel:card0.131.CPU3 0 1523264148
yuri.cpu.PCI-MSI.524288-edge.nvidia.132.CPU0 0 1523264148
yuri.cpu.PCI-MSI.524288-edge.nvidia.132.CPU1 6145 1523264148
yuri.cpu.PCI-MSI.524288-edge.nvidia.132.CPU2 436286 1523264148
yuri.cpu.PCI-MSI.524288-edge.nvidia.132.CPU3 0 1523264148
```
```
bundle exec metrics-cpu-softirqs.rb  
yuri.cpu.HI.CPU0 3 1523266763
yuri.cpu.HI.CPU1 8 1523266763
yuri.cpu.HI.CPU2 3 1523266763
yuri.cpu.HI.CPU3 2 1523266763
yuri.cpu.TIMER.CPU0 178041 1523266763
yuri.cpu.TIMER.CPU1 193639 1523266763
yuri.cpu.TIMER.CPU2 183581 1523266763
yuri.cpu.TIMER.CPU3 192156 1523266763
yuri.cpu.NET_TX.CPU0 284 1523266763
yuri.cpu.NET_TX.CPU1 3 1523266763
yuri.cpu.NET_TX.CPU2 4 1523266763
yuri.cpu.NET_TX.CPU3 8 1523266763
yuri.cpu.NET_RX.CPU0 5150 1523266763
yuri.cpu.NET_RX.CPU1 350 1523266763
yuri.cpu.NET_RX.CPU2 1906 1523266763
yuri.cpu.NET_RX.CPU3 66321 1523266763
yuri.cpu.BLOCK.CPU0 163878 1523266763
yuri.cpu.BLOCK.CPU1 68 1523266763
yuri.cpu.BLOCK.CPU2 21183 1523266763
yuri.cpu.BLOCK.CPU3 24082 1523266763
yuri.cpu.IRQ_POLL.CPU0 0 1523266763
yuri.cpu.IRQ_POLL.CPU1 0 1523266763
yuri.cpu.IRQ_POLL.CPU2 0 1523266763
yuri.cpu.IRQ_POLL.CPU3 0 1523266763
yuri.cpu.TASKLET.CPU0 56 1523266763
yuri.cpu.TASKLET.CPU1 110 1523266763
yuri.cpu.TASKLET.CPU2 2895 1523266763
yuri.cpu.TASKLET.CPU3 1509 1523266763
yuri.cpu.SCHED.CPU0 163085 1523266763
yuri.cpu.SCHED.CPU1 178888 1523266763
yuri.cpu.SCHED.CPU2 159592 1523266763
yuri.cpu.SCHED.CPU3 166079 1523266763
yuri.cpu.HRTIMER.CPU0 0 1523266763
yuri.cpu.HRTIMER.CPU1 0 1523266763
yuri.cpu.HRTIMER.CPU2 0 1523266763
yuri.cpu.HRTIMER.CPU3 0 1523266763
yuri.cpu.RCU.CPU0 231638 1523266763
yuri.cpu.RCU.CPU1 227187 1523266763
yuri.cpu.RCU.CPU2 257133 1523266763
yuri.cpu.RCU.CPU3 231924 1523266763
```
```
bundle exec metrics-softnet-stat.rb            
yuri.cpu.processed.CPU0 88095 1523271182
yuri.cpu.dropped.CPU0 0 1523271182
yuri.cpu.time_squeeze.CPU0 674 1523271182
yuri.cpu.cpu_collision.CPU0 0 1523271182
yuri.cpu.received_rps.CPU0 0 1523271182
yuri.cpu.flow_limit_count.CPU0 0 1523271182
yuri.cpu.processed.CPU1 1860 1523271182
yuri.cpu.dropped.CPU1 0 1523271182
yuri.cpu.time_squeeze.CPU1 6 1523271182
yuri.cpu.cpu_collision.CPU1 0 1523271182
yuri.cpu.received_rps.CPU1 0 1523271182
yuri.cpu.flow_limit_count.CPU1 0 1523271182
yuri.cpu.processed.CPU2 30160 1523271182
yuri.cpu.dropped.CPU2 0 1523271182
yuri.cpu.time_squeeze.CPU2 200 1523271182
yuri.cpu.cpu_collision.CPU2 0 1523271182
yuri.cpu.received_rps.CPU2 0 1523271182
yuri.cpu.flow_limit_count.CPU2 0 1523271182
yuri.cpu.processed.CPU3 69363 1523271182
yuri.cpu.dropped.CPU3 0 1523271182
yuri.cpu.time_squeeze.CPU3 446 1523271182
yuri.cpu.cpu_collision.CPU3 0 1523271182
yuri.cpu.received_rps.CPU3 0 1523271182
yuri.cpu.flow_limit_count.CPU3 0 1523271182
```

### CODE REVIEW

File: bin/metrics-cpu-pcnt-usage.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@ def acquire_proc_stats
       name = info.shift
 
       # we are matching TOTAL stats and returning a hash of values
-      if name =~ /^cpu$/
+      if name.match?(/^cpu$/)
```
Comment (majormoses) [inline bin/metrics-cpu-pcnt-usage.rb]: `.match?` was added in ruby 2.4

File: bin/metrics-cpu.rb [hunk 2]
```diff
@@ -50,7 +50,7 @@ def run
       next if info.empty?
       name = info.shift
 
-      if name =~ /cpu([0-9]+|)/
+      if name.match?(/cpu([0-9]+|)/)
```
Comment (majormoses) [inline bin/metrics-cpu.rb]: `.match?` was added in ruby 2.4

**Review by majormoses — CHANGES_REQUESTED** [review]
Overall I think it looks good but has some unintentional breaking changes.

**Comment by yuri-zubov** [comment]
@majormoses - fixed. Recheck, please

**Review by majormoses — APPROVED** [review]

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-cpu-checks/versions/2.1.0

**Pull request merged**